File: bench/cases/fifo-queue-100k/main.ts
const q: number[] = []
let total = 0
const n = 100000
for (let i = 0; i < n; i = i + 1) {
  q.push(i)
  if (q.length > 16) {
    total = total + q.shift()!
  }
}
while (q.length > 0) {
  total = total + q.shift()!
}
console.log(total)

File: bench/cases/fifo-queue-100k/main.rs
use std::collections::VecDeque;

fn main() {
    let mut q: VecDeque<i64> = VecDeque::new();
    let mut total: i64 = 0;
    let n: i64 = 100_000;
    for i in 0..n {
        q.push_back(i);
        if q.len() > 16 {
            total += q.pop_front().unwrap();
        }
    }
    while let Some(v) = q.pop_front() {
        total += v;
    }
    println!("{}", total);
}
